### ISSUE TITLE
[CKEditor 5] `TypeError: e.matchAll` when using MathType in combination with Source Editing

### ISSUE THREAD

**Issue by mabryl** [wiris/html-integrations#868]
## Description

In an editor build that includes the MathType plugin alongside [Source Editing](https://ckeditor.com/docs/ckeditor5/latest/features/source-editing.html), the editor will throw the following error when any changes are made in the source editing mode:

```
Uncaught TypeError: e.matchAll is not a function or its return value is not iterable
```

## Environment

@wiris/mathtype-ckeditor5": "8.7.2

*What is the relevant software and their versions?*

- *Editor (CKEditor, Froala, TinyMCE, etc.)* - CKEditor 5 v40.2.0
- *Framework (Angular, React, etc.)* - vanilla JS
- *Browser (Chrome, Firefox, etc.)* - cross-browser
- *Operating System (Windows, macOS, Ubuntu, etc.)* - tested on macOS Sonoma 14.1.2 but most likely cross-OS

## Steps to reproduce

1. Open the demo editor that's attached at the bottom of the ticket
2. Enter source editing mode by pressing the "Source" button on the toolbar
3. Make any change to the content
4. Try exiting the source editing mode

Video for reference:

https://github.com/wiris/html-integrations/assets/72079603/0e31067d-94b9-40bf-8d14-9e2f0f8034aa

## Expected result

You should be able to exit source editing mode.

## Actual result

The editor freezes, an error is thrown in the console.

## Other details

This error is reproducible only when using the MathType plugin. When you remove the plugin from the editor build and rebuild it, the issue is not reproducible.

Demo editor available below. To run it, spin up a dev server (e.g. with [http-server](https://www.npmjs.com/package/http-server)) and access the `index.html` file in the `sample` directory.

[ckeditor5-only-math-and-source.zip](https://github.com/wiris/html-integrations/files/13727552/ckeditor5-only-math-and-source.zip)

**Comment by carla-at-wiris** [comment]
Hi @mabryl, thank you for reporting the issue.

We are working on providing a fix. Once we release a new version compatible with the **Source Editing** CKEditor 5 plugin, we will let you know.

**Comment by jpgupta** [comment]
Hi @carla-at-wiris - thanks for team's work on this, we're running into the same issue. 

It looks like a fix is available in 8.8.2 - we're keen to use this however I noticed that as a part of this fix, you've also bumped the minimum required version of CKEditor by quite a few major releases. 

Just wanted to know is this necessary for the particular fix? If so then we'll have to upgrade our version of CKEditor which is fine but i'm hoping that this particular fixed could be done in a semver-friendly way?

**Comment by carla-at-wiris** [comment]
Hi @jpgupta, the issue is fixed in the latest release version `8.8.2`. 

We updated the CKEditor5 dependencies in order to use their latest fixes and upgrades. It's not necessary to use those versions, the fix works with previous CKEditor5 releases.

**Comment by jpgupta** [comment]
@carla-at-wiris - ok thanks for the quick reply!
